### ISSUE TITLE
Refactor "internal documentation" and "Release"

### ISSUE THREAD

**Issue by fvisin** [Theano/Theano#4075]
Part of the information in "Internal documentation" could be spread into other sections of the documentation (e.g. the python booster can go into the "Prerequisites" or "Further readings" of the tutorial) and the rest should be moved to the internal MILA website IMO. 

I also think that "Release" could be moved to the internal website.

**Comment by yash14123** [comment]
Fixing this one in the following PR.

**Comment by fvisin** [comment]
I don't think you have access to our internal website, so you probably cannot fix this one.
